source/functional-concepts.md

```diff
@@ -0,0 +1,150 @@
+# Functional Programming in Swift: A Primer
+
+Swift is a hybrid language that allows both objective-oriented and functional code to co-exist.
+
+### First-class functions
+
+First-class functions are *values* that can either take other functions as arguments or return them as results. Also refered to as *higher order functions*.
```

Comment:
Well in the least, can you not say that a first-class function is a higher-order function? Glossing over things is fine but I'd rather not have factually incorrect things.